backend/sportsmatch/src/main/java/com/sportsmatch/controllers/SportController.java

```diff
@@ -0,0 +1,30 @@
+package com.sportsmatch.controllers;
+
+import com.sportsmatch.dtos.SportDTO;
+import com.sportsmatch.service.SportService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/v1/sports")
+public class SportController {
+
+    private final SportService sportService;
+
+    /**
+     * This endpoint returns paginated list of SportDto.
+     *
+     * @param pageable contains tha page and size value
+     * @return paginated list of SportDTO
+     */
+    @GetMapping("/all")
+    public List<SportDTO> getSports(final Pageable pageable) {
```

Comment:
I realised that we might not need pagination for sports because there will probably only be at most 50 different sports. But lets keep it for now as an example

Reply:
I miss understood it sorry. I thought later there will be someting filter for it too like search in it or something like that